int32 size_x
int32 size_y
float64[] map
